AppDeployToolkit/AppDeployToolkitExtensions.ps1

```diff
@@ -3982,15 +4027,27 @@ function Get-NxtFileEncoding {
 		[string]${cmdletName} = $PSCmdlet.MyInvocation.MyCommand.Name
 	}
 	Process {
-		try {
-			[string]$intEncoding = [PSADTNXT.Extensions]::GetEncoding($Path)
-			if ($true -eq ([System.String]::IsNullOrEmpty($intEncoding))) {
-				[string]$intEncoding = $DefaultEncoding
+		if ($false -eq (Test-Path -Path $Path)) {
```

Comment:
According to param description instead of testing the file exists case I'd rather add the write output to the catch block. A seperate check is not really necessary, is it?

Reply:
Need more details to understand the request

Reply:
spare this for another issue if necessary

Reply:
Instead of the test above this would suffice:

```
try {
	[string]$intEncoding = [PSADTNXT.Extensions]::GetEncoding($Path)
	if ($true -eq ([System.String]::IsNullOrEmpty($intEncoding))) {
		Write-Log "Failed to detect encoding for file [$Path]. Using default encoding [$DefaultEncoding]." -Severity 2 -Source ${cmdletName}
		[string]$intEncoding = $DefaultEncoding
	}
	Write-Output $intEncoding
}
catch {
	Write-Log -Message "Failed to run the encoding detection. Default encoding [$DefaultEncoding] is returned.`n$(Resolve-Error)" -Severity 3 -Source ${cmdletName}
	Write-Output $DefaultEncoding
}
```

Reply:
An Exception from [PSADTNXT.Extensions]::GetEncoding($Path) is not the same as an unspecified encoding, wont implement this change prior to further reasons